presto-main/src/main/java/com/facebook/presto/execution/StageInfo.java

```diff
@@ -126,4 +126,15 @@ public String toString()
     {
         return stageInfo.map(StageInfo::getAllStages).orElse(ImmutableList.of());
     }
+
+    public Optional<StageInfo> getStageWithStageId(StageId stageId)
```

Comment:
might be a good idea to add some unit tests for this method specifically.

Reply:
Added